pkg/services/blockchain/eth/resolver.go

```diff
@@ -124,6 +194,29 @@ func (r *Resolver) Resolve(
 	default:
 		stateInfo, gistInfo, err = r.resolveLatest(ctx, userID)
 	}
+	// err == http.505
+	if err != nil && (!errors.Is(err, services.ErrNotFound) || opts.GistRoot != nil) {
```

Comment:
Why we check if GetGist not nil? it seems to me that the previous version had the following check:
```go
if err != nil && !errors.Is(err, services.ErrNotFound) { return err }
```

Reply:
I found some case where state not found but it should return some info making tests when opts.State !=nil

Reply:
Could you share the case to test, please?

Reply:
Removed finally